contrib/babelfishpg_tsql/src/pltsql_coerce.c

```diff
@@ -998,6 +1004,75 @@ get_immediate_base_type_of_UDT_internal(Oid typeid)
 	return base_type;
 }
 
+/*
+ * get_immediate_base_typmod_of_UDT_internal()
+ * This function returns typmod of the Immediate base type for UDT.
+ * is_UDT helps to know, weather the typeid is of UDT or not.
+ * This function also includes special handling of sys.sysname,
+ * that is a domain created from VARCHAR(128) in BBF.
+ * By default, typmod of sysname is SYSNAME_TYPMOD
+ */
+static int32
+get_immediate_base_typmod_of_UDT_internal(Oid typeid, bool *is_UDT)
```

Comment:
Instead of this lets use `getBaseTypeAndTypmod` function to fetch the typmod, which will work for all UDT, sysname and UDT on top of sysname datatype.

Reply:
Made use of getBaseTypeAndTypmod function to fetch typmod.